addons/link_tracker/models/mail_render_mixin.py

```diff
@@ -38,6 +38,9 @@ def _shorten_links(self, html, link_tracker_vals, blacklist=None, base_url=None)
         short_schema = base_url + '/r/'
         for match in set(re.findall(tools.HTML_TAG_URL_REGEX, html)):
             long_url = match[1]
+            # Make relative links absolute
+            if long_url.startswith(('/', '?', '#')):
```

Comment:
Maybe parse the url and check netloc so we don't miss any weird cases?

Reply:
netloc is not reliable enough
```py
url = url_parse('example.com')
url.netloc  # ""
url.path # "example.com"
```
the only weird case I can think of is the case where the link start with spaces but they are trimmed by the webclient already, and for xmlrpc the people on the other side are responsible for their inputs